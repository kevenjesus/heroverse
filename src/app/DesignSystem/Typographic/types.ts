import { AlignType, colorsType, fontSizeType, weightTye } from "@/app/styles/types";
import { ReactNode } from "react";

export interface TypographicProps {
    children: ReactNode
    size?: fontSizeType
    weight?: weightTye
    color?: colorsType
    align?: AlignType
}