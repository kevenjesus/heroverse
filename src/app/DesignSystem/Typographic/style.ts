import { colors, fontSizes, weightValues } from '@/app/styles/guide'
import { FontStyleType } from '@/app/styles/types'
import styled from 'styled-components'

export const TypographicItem = styled.p<FontStyleType>`
    font-size: ${({size}) => size ? size : fontSizes.regular};
    font-weight: ${({weight}) => weight ? weight : weightValues.regular};
    color: ${({color}) => color ? color : colors.additional.black};
`