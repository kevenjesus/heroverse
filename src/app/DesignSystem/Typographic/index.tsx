import * as S from './style'
import { TypographicProps } from './types'

export default function Typographic({children, ...rest}: TypographicProps) {
    return (
        <S.TypographicItem {...rest}>{children}</S.TypographicItem>
    )
}