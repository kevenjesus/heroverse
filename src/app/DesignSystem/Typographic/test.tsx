import { render } from '@testing-library/react';
import Typographic from '.';
import '@testing-library/jest-dom'

describe('Typographic', () => {
  it('should render', () => {
    const { container } = render(<Typographic>heroverse</Typographic>);
    expect(container).toMatchSnapshot()
  });

  it('should render props correctly', () => {
    const { getByText } = render(<Typographic color='darkBlue' size='20px' weight='700'>heroverse</Typographic>);
    expect(getByText('heroverse')).toBeInTheDocument()
  });
});
