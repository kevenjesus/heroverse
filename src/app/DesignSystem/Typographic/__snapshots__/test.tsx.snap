// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typographic should render 1`] = `
<div>
  <p
    class="sc-dmqHEX hZDiJS"
  >
    heroverse
  </p>
</div>
`;
